org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/SeekTest.java

```diff
@@ -0,0 +1,146 @@
+import org.eclipse.scanning.api.scan.event.IPositionListener;
+import org.eclipse.scanning.api.scan.models.ScanModel;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class SeekTest extends AbstractAcquisitionTest {
+
+	@Before
+	public void setupServices() throws Exception {
+		super.setupServices();
+	}
+
+	@Test
+	public void seekFirst() throws Exception {
+		
+		IDeviceController controller = createTestScanner(null);
+		AbstractRunnableDevice<ScanModel> scanner = (AbstractRunnableDevice<ScanModel>)controller.getDevice();
+    
+		try {
+			scanner.start(null);		
+			Thread.sleep(100);	
+			scanner.pause();
+			
+			IPosition first   = scanner.getModel().getPositionIterable().iterator().next();
+			IPosition current = scanner.getPositioner().getPosition();
+			assertNotNull(current);
+			assertNotEquals(first, current);
+			
+			scanner.seek(0);
+			current = scanner.getPositioner().getPosition();
+			assertNotNull(current);
+			assertEquals(first.getStepIndex(), current.getStepIndex());
+			
+		} finally {
+			scanner.abort();
+		}
+	}
+	
+	@Ignore("This MUST pass but requires a larger refactor so seeing that the initial changes do with the tests first.")
+	@Test
+	public void seekFirstRestartsInCorrectLocation() throws Exception {
+		
+		IDeviceController controller = createTestScanner(null);
+		AbstractRunnableDevice<ScanModel> scanner = (AbstractRunnableDevice<ScanModel>)controller.getDevice();
+		
+		final List<Integer> steps = new ArrayList<Integer>();
+		scanner.addPositionListener(new IPositionListener() {
+			public void positionPerformed(PositionEvent evt) throws ScanningException {
+				steps.add(evt.getPosition().getStepIndex());
+			}
+		});
+		try {
+			scanner.start(null);		
+			Thread.sleep(100);	
+			scanner.pause();
+			
+			IPosition first   = scanner.getModel().getPositionIterable().iterator().next();
+			IPosition stopped = scanner.getPositioner().getPosition();
+			assertNotNull(stopped);
+			assertNotEquals(first, stopped);
+			assertTrue(stopped.getStepIndex()>0);
+			assertTrue(stopped.getStepIndex()<24);
+			
+			scanner.seek(0);
+			IPosition current = scanner.getPositioner().getPosition();
+			assertNotNull(current);
+			assertEquals(first.getStepIndex(), current.getStepIndex());
+			scanner.resume();
+			scanner.latch(10, TimeUnit.SECONDS);
+			
+			// The scan should restart from where it is seeked to.
+			// Therefore the steps should be size (25) + stopped.getStepIndex()
+			assertEquals("The scan should restart from where it is seeked to.",
+					     25+stopped.getStepIndex(), steps.size());
+			
+		} finally {
+			scanner.abort();
+		}
+	}
+
+	@Test(expected=ScanningException.class)
+	public void seekTooLarge() throws Exception {
+
+		IDeviceController controller = createTestScanner(null);
+		AbstractRunnableDevice<ScanModel> scanner = (AbstractRunnableDevice<ScanModel>)controller.getDevice();
+    
+		try {
+			scanner.start(null);		
+			Thread.sleep(100);	

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)